### ISSUE TITLE
ContentAssistTest for Template Proposals did not work

### ISSUE THREAD

**Issue by miklossy** [eclipse/xtext-eclipse#219]
The following Content Assist test case on the org.eclipse.xtext.example.domainmodel project fails, but it should succeed.

![templateproposaltest](https://cloud.githubusercontent.com/assets/20393472/24425728/79ec0dd0-1405-11e7-85c6-62a9be0222da.png)

Reproducible with Eclipse Neon.2 and Xtext 2.11.0.

**Comment by cdietrich** [comment]
problem is: test calls `org.eclipse.jface.text.templates.TemplateProposal.apply(IDocument)` which is a noop at least in neon+

**Comment by cdietrich** [comment]
workaround

```
class ContentAssistTest extends AbstractContentAssistTest {
	
	@Inject
	Injector injector
	
	override protected newBuilder() throws Exception {
		new ContentAssistProcessorTestBuilder(injector, this) {
			
			override protected applyProposal(ICompletionProposal proposal, IXtextDocument document) throws Exception {
		val Shell shell = new Shell();
		try {
			val XtextSourceViewerConfiguration configuration = injector.getInstance(XtextSourceViewerConfiguration);
			val ISourceViewer sourceViewer = getSourceViewer(shell, document, configuration);
			return appendAndApplyProposal(proposal, sourceViewer, model, document.length);
		} finally {
			shell.dispose();
		}
			}
			
		}
	}
```

**Comment by cdietrich** [comment]
Fixed in 2.12
